PlayerSkeleton.java

```diff
@@ -1,22 +1,217 @@
+			boolean full = true;
+			for(int c = 0; c < col; c++) {
+				if(field[r][c] == 0) {
+					full = false;
+					break;
+				}
+			}
+			//if the row was full - remove it and slide above stuff down
+			if(full) {
+				rowsCleared++;
+				//for each column
+				for(int c = 0; c < col; c++) {
+
+					//slide down all bricks
+					for(int i = r; i < top[c]; i++) {
+						field[i][c] = field[i+1][c];
+					}
+					//lower the top
+					top[c]--;
+					while(top[c]>=1 && field[top[c]-1][c]==0) {
+						top[c]--;
+					}
+				}
+			}
+		}
+
+		return new Pair(field, rowsCleared);
+	}
+
+	/**
+	* This method prints current board configuration
+	* @param currentField
+	**/
+	private void printField(int[][] currentField) {
+		System.out.println("========Field==============");
+		for(int i=0; i<currentField.length; i++) {
+			for(int j=0; j<currentField[i].length; j++) {
+				System.out.print(currentField[i][j]);
+			}
+			System.out.println();
+		}
+	}
+
+	/**
+	* This method gets the information about a piece's shape from state s.
+	* @Param s (current board configuration)
+	**/
+	private void getPieceInformation(State s) {
+		pBottom = s.getpBottom();
+		pTop = s.getpTop();
+		pHeight = s.getpHeight();
+		pWidth = s.getpWidth();
+	}
+
+	/**
+	* This method makes a copy of an int array.
+	* @param originalArray (array to copy)
+	* @return newArray (deep copy of the array)
+	**/
+	private int[] copyArray(int[] originalArray) {
+		if(originalArray == null) {
+			return null;
+		}
+
+		int[] newArray = Arrays.copyOf(originalArray, originalArray.length);
+		return newArray;
+	}
+
+	/**
+	* This method makes a copy of an 2D int array.
+	* @param originalArray (array to copy)
+	* @return newArray (deep copy of the array)
+	**/
+	private int[][] copyArray(int[][] originalArray) {
+		if(originalArray == null) {
+			return null;
+		}
+
+		int[][] newArray = new int[originalArray.length][];
+		for(int i=0; i<originalArray.length; i++) {
+			newArray[i] = Arrays.copyOf(originalArray[i], originalArray[i].length);
+		}
+		return newArray;
+	}
+
+	/**
+	* This method makes a copy of an 3D int array.
+	* @param originalArray (array to copy)
+	* @return newArray (deep copy of the array)
+	**/
+	private int[][][] copyArray(int[][][] originalArray) {
+		if(originalArray == null) {
+			return null;
+		}
+
+		int[][][] newArray = new int[originalArray.length][][];
+		for(int i=0; i<originalArray.length; i++) {
+			newArray[i] = new int[originalArray[i].length][];
+			for(int j=0; j<originalArray[i].length; j++) {
+				newArray[i][j] = Arrays.copyOf(originalArray[i][j], originalArray[i][j].length);
+			}
+		}
+		return newArray;
 	}
 	
 	public static void main(String[] args) {
 		State s = new State();
 		new TFrame(s);
 		PlayerSkeleton p = new PlayerSkeleton();
+		p.getPieceInformation(s);
 		while(!s.hasLost()) {
 			s.makeMove(p.pickMove(s,s.legalMoves()));
 			s.draw();
 			s.drawNext(0,0);
 			try {
-				Thread.sleep(300);
+				Thread.sleep(1500);
```

Comment:
How come?

Reply:
This have to be changed back, I was testing as it was too fast. My bad